htlcswitch/hop/payload.go

```diff
@@ -277,6 +286,32 @@ func ValidateParsedPayloadTypes(parsedTypes tlv.TypeMap,
 	needNextHop := !(hasEncryptedData || isFinalHop)
 
 	switch {
+	// Both blinding point being set is invalid.
+	case hasBlinding && updateAddBlinding:
+		return ErrInvalidPayload{
+			Type:      record.BlindingPointOnionType,
+			Violation: IncludedViolation,
+			FinalHop:  isFinalHop,
+		}
+
+	// If encrypted data is not provided, blinding points should not be
+	// set.
+	case !hasEncryptedData && (hasBlinding || updateAddBlinding):
+		return ErrInvalidPayload{
+			Type:      record.EncryptedDataOnionType,
+			Violation: OmittedViolation,
```

Comment:
Reason for framing it this way is that we don't have a `Type` for the case where `updateAddBlinding` is set and `!hasEncryptedData` because the blinding point isn't from the payload itself, so didn't know how to frame it in the context of a payload error.

We _could_ have a `IncludedViolation` for with `record.BlindingPointOnionType` but that's kinda odd if it's _actually_ the `UpdateAddHTLC` blinding point is set, so I thought this was a lesser evil.

Reply:
Ah I see, I didn't pay attention to the type :+1: